### PR TITLE
feat(Observable): add pairwise operator

### PR DESCRIPTION
**Pull request by justinwoo**
bring in pairwise operator from [RxJS4](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/pairwise.md).

Uses the same "has previous" and "previous" value with new array each time approach.

edit: @Frikki said that this is pretty much `bufferCount` but with a static buffer size, so maybe this isn't needed. At least pairwise will be faster!!!! Maybe. I use pairwise in RxJS4 :frowning:

### CODE REVIEW

**Comment by trxcllnt** [comment]
@justinwoo thanks for the PR! I agree, a specific implementation for `pairwise` will be faster than `bufferCount`, and probably easier to understand when you're using it. @blesh +1 for merging.

**Comment by kwonoj** [comment]
Change looks fine to me too in general, maybe we can consider to place this under `kitchenSink`? just my 2¢ opinion.

**Comment by blesh** [comment]
@justinwoo ... great PR. This looks good so far, however I'd like a few things before I can merge it:

1. Tests covering: `empty`, `never`, error (an error mid-stream), and `throw` (Observable.throw).
2. A Test covering an observable that only emits one value.
3. Probably remove this from CoreOperators and Rx.ts, and just leave it in KitchenSink. 
4. A micro perf test vs RxJS 4's impl (to make sure we're doing the right thing)
4. Some inline docs.

We'll be publishing beta in short order. In light of that, I'd like any new operators to be very well covered with tests.

**Comment by justinwoo** [comment]
Turns out DFW has good wifi :open_mouth: 

Updated to use the KitchenSink type (using similar definition to `elementAt`) and added tests for `empty`, `never`, and `throw` using the marble diagram tests.

File: src/Rx.ts [hunk 1]
```diff
@@ -65,6 +65,7 @@ import './add/operator/mergeMap';
 import './add/operator/mergeMapTo';
 import './add/operator/multicast';
 import './add/operator/observeOn';
+import './add/operator/pairwise';
```
Comment (kwonoj) [inline src/Rx.ts]: This would not be required if this operator's placed under `kitchenSink`, since importing `add` will patch prototype immediately.

Comment (justinwoo) [inline src/Rx.ts]: Oh, oops, accidentally unstaged this.

File: spec/operators/pairwise-spec.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    var values = {
+    };
+
+    var source = e1.pairwise();
+
+    expectObservable(source).toBe(expected, values);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+  });
+
+  it('handle mid-stream throw', function () {
```
Comment (kwonoj) [inline spec/operators/pairwise-spec.js]: I know it's quite minor thing, would you able to align test case names as same as others? (starts with `should..`)

**Comment by kwonoj** [comment]
Change looks good to me.

File: spec/operators/pairwise-spec.js (hunk 2 above)
Comment (justinwoo) [inline spec/operators/pairwise-spec.js]: Sure, pushed the change now.

**Comment by blesh** [comment]
Merged with 1432e59... great work, @justinwoo!

**Comment by justinwoo** [comment]
:+1: :beers:

**Pull request closed without merging**